src/Core/src/Platform/IMauiContext.cs

```diff
@@ -11,7 +11,21 @@ public interface IMauiContext
 
 #if __ANDROID__
 		global::Android.Content.Context? Context { get; }
-#elif __IOS__
+#endif
+	}
+
+	public interface IMauiApplicationContext : IMauiContext
+	{
+#if __IOS__
+		UIKit.UIApplicationDelegate? ApplicationDelegate { get; }
+#elif WINDOWS
+		UI.Xaml.Application? Application { get; }
+#endif
+	}
+
+	public interface IMauiWindowContext : IMauiContext
```

Comment:
A new `IMauiWindowContext` that is basically used everywhere except for applications and has UI things, such as windows, timers and window managers.